lib/maths/unittest/CRandomizedPeriodicityTestTest.cc

```diff
@@ -0,0 +1,202 @@
+
+        for (std::size_t i = 0u; i < samples.size(); ++i) {
+            for (std::size_t j = 0u; j < boost::size(functions); ++j) {
+                rtests[j].add(time, 600.0 * (functions[j])(time) + samples[i]);
+            }
+            if (time >= day + DAY) {
+                for (std::size_t j = 0u; j < boost::size(rtests); ++j) {
+                    if (j < 3) {
+                        (rtests[j].test() ? falsePositives[j] : trueNegatives[j]) += 1.0;
+                    } else {
+                        (rtests[j].test() ? truePositives[j - 3]
+                                          : falseNegatives[j - 3]) += 1.0;
+                        if (rtests[j].test()) {
+                            timeToDetectionMoments[j - 3].add(
+                                time - lastTruePositive[j - 3]);
+                            timeToDetectionMax[j - 3].add(
+                                static_cast<double>(time - lastTruePositive[j - 3]));
+                            lastTruePositive[j - 3] = time;
+                        }
+                    }
+                }
+                day += DAY;
+            }
+            time += HALF_HOUR;
+        }
+
+        LOG_DEBUG(<< "falsePositives = " << core::CContainerPrinter::print(falsePositives));
+        LOG_DEBUG(<< "trueNegatives = " << core::CContainerPrinter::print(trueNegatives));
+        for (std::size_t i = 0u; i < boost::size(falsePositives); ++i) {
+            CPPUNIT_ASSERT(falsePositives[i] / trueNegatives[i] < 0.1);
+            typeI.add(falsePositives[i] / trueNegatives[i]);
+        }
+        LOG_DEBUG(<< "truePositives = " << core::CContainerPrinter::print(truePositives));
+        LOG_DEBUG(<< "falseNegatives = " << core::CContainerPrinter::print(falseNegatives));
+        for (std::size_t i = 0u; i < boost::size(falsePositives); ++i) {
+            CPPUNIT_ASSERT(falseNegatives[i] / truePositives[i] < 0.2);
+            typeII.add(falseNegatives[i] / truePositives[i]);
+        }
+
+        for (std::size_t i = 0u; i < boost::size(timeToDetectionMoments); ++i) {
+            LOG_DEBUG(<< "time to detect moments = " << timeToDetectionMoments[i]);
+            LOG_DEBUG(<< "maximum time to detect = " << timeToDetectionMax[i][0]);
+            CPPUNIT_ASSERT(maths::CBasicStatistics::mean(timeToDetectionMoments[i]) <
+                           1.5 * DAY);
+            CPPUNIT_ASSERT(std::sqrt(maths::CBasicStatistics::variance(
+                               timeToDetectionMoments[i])) < 5 * DAY);
+            CPPUNIT_ASSERT(timeToDetectionMax[i][0] <= 27 * WEEK);
+        }
+    }
+    LOG_DEBUG(<< "type I  = " << maths::CBasicStatistics::mean(typeI));
+    LOG_DEBUG(<< "type II = " << maths::CBasicStatistics::mean(typeII));
+    CPPUNIT_ASSERT(maths::CBasicStatistics::mean(typeI) < 0.015);
+    CPPUNIT_ASSERT(maths::CBasicStatistics::mean(typeII) < 0.05);
+}
+
+void CRandomizedPeriodicityTestTest::testPersist() {
+    // Check that persistence is idempotent.
```

Comment:
this test only compares XML and the rng. It would be good if it could actually test the actual class instances. 

I see that `checksum` being disabled(unrelated for this PR), but maybe you can checksum some of the members and compare them. Should be possible as you have `friend` access.

Reply:
In fact, I was able to re-enable the checksum implementation because of changes I'd made to `maths::CTimeSeriesDecomposition`.